### DISCUSSION TITLE
using of --cache option seems to do nothing

### DISCUSSION THREAD

**Discussion by koukalp** [eslint/eslint#15335] [Q&A] [answered]
Dear all, 

Sorry for my dummy question, however, I didn't find the answer anywhere (including eslint docs). 

**How do I use `--cache` option in CLI?** 

I'd expect I could run `eslint . --cache` which means:
1. eslint will be run in the current directory => OK
2. `.eslintrc.*` file will be loaded for additional configuration => OK
3. eslint will check `.eslintcache` file. If it exists, it will check only changed files. If it doesn't exists, it will check all files
4. a new `.eslintcache` file will be created in the directory where eslint was run

My problem is, that points 3 and 4 I never see in action. No cache file is ever created anywhere (even if a provide `--cache-location .mycache`). 
- I am sure there are files being linted
- I tried running towards files that have no errors / warnings
- I tried running towards mixture of files with and without errors / warning

Using eslint v8.2.0, node 12.22.7, npm 8.3.1

Thank you for your anwers!

**Comment by nzakas** [comment, marked as answer]
That is the way it should work. Note if you run the CLI with —cache and then run it again without —cache, the cache file is deleted.

The best way to see what’s going on is to run ESLint with —debug. That should tell you if anything strange is happening.

**Reply by koukalp** [reply]
Thank you for your confirmation. Since using `--debug` flag didn't work either, I realised I run the following command which basically ignored all options that were passed to `eslint`. Since I don't even need `dotenv`, removing it from the command fixed my issue.

`dotenv test.env eslint . --cache` -> `eslint . --cache`

Thank you! Your answer made me realise what I was doing wrong
